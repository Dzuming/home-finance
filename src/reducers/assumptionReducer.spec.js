import deepFreeze from 'deep-freeze';
import reducer from './budgetReducer';
import {
  setAssumption,
  setOverallAssumptions,
} from '../actions/assumptionActions';

const initialState = deepFreeze(reducer({}, { type: 'INIT' }));
const assumptions = [
  {
    id: 1,
    name: 'Poduszka bezpieczeństwa',
    percentage: 20,
    value: 66.6,
  },
  {
    id: 2,
    name: 'Wakacje',
    percentage: 10,
    value: 33.3,
  },
  {
    id: 3,
    name: 'Wakacje',
    percentage: 50,
    value: 695,
  },
];

const overallAssumptions = [
  {
    name: 'Poduszka bezpieczeństwa',
    value: 66.6,
  },
  {
    name: 'Wakacje',
    value: 33.3,
  },
  {
    name: 'Wakacje',
    value: 695,
  },
];

describe('assumption reducer', () => {
  it('should handle unknown actions', () => {
    expect(reducer(initialState, { type: 'FAKE' })).toBe(initialState);
  });

  describe('get assumption', () => {
    it('should get assumption', () => {
      expect(
        reducer(initialState, setAssumption(assumptions)),
      ).toMatchSnapshot();
    });
    it('should get overall assumption', () => {
      expect(
        reducer(initialState, setOverallAssumptions(overallAssumptions)),
      ).toMatchSnapshot();
    });
  });
});
