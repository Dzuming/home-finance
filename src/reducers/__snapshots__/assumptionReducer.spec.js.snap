// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assumption reducer get assumption should get assumption 1`] = `Object {}`;
